src/Tribe/Views/V2/Template/Excerpt.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Handles the manipulation of the excerpt.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Template
+ */
+
+namespace Tribe\Events\Views\V2\Template;
+use Tribe__Template as Base_Template;
+use Tribe__Events__Main as Plugin;
+
+/**
+ * Class Excerpt
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Template
+ */
+class Excerpt extends Base_Template {
+
+	/**
+	 * Excerpt constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->set_template_origin( Plugin::instance() );
+		$this->set_template_folder( 'src/views/v2' );
+		$this->set_template_context_extract( true );
+		$this->set_template_folder_lookup( true );
+	}
+
+	/**
+	 * Filters the excerpt length.
+	 *
+	 * Set the excerpt length for list and day view.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $length The excerpt length.
+	 *
+	 * @return int The excerpt length modified, if necessary.
+	 */
+	public function maybe_filter_excerpt_length( $length ) {
+		$context = tribe_context();
+		$view = $context->get( 'event_display_mode', 'list' );
+
+		return in_array( $view, [ 'list', 'day' ] ) ? 30 : $length;
+	}
+
+	/**
+	 * Filters the excerpt more button.
+	 *
+	 * Set the excerpt more button styles for twentyseventeen.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $link The excerpt read more link.
+	 *
+	 * @return string The excerpt read more link modified, if necessary.
+	 */
+	public function maybe_filter_excerpt_more( $link ) {
+
+		if ( is_admin() ) {
+			return $link;
+		}
+
+		$template = strtolower( get_template() );
+
+		// Check if theme is twentyseventeen.
+		if ( ! $template || 'twentyseventeen' !== $template ) {
+			return $link;
+		}
+
+		return $this->template( 'components/read-more', [], false );
```

Comment:
We should pass down to the template `'link' => $link` and instead of using `get_the_ID()` on the actual template we should be passing `'event' => tribe_get_event( get_the_ID() )`, so it more closely matches the other templates.

Don't forget to add these two on the docblock of the new view.

Reply:
Thanks! I don't understand why is that we should be sending the $link variable. That variable contains what twentyseventeen set as a string for the whole "continue reading". And what we're trying to do here is basically override that with our template. We don't need any info from that variable in the template: https://github.com/WordPress/twentyseventeen/blob/master/functions.php#L222

Reply:
And actually, if we use tribe_get_event there it end up in a fatal.

Reply:
What I meant is that we need to pass into the template the variables in case someone wants to use that in the template. Why do we end up with a fatal? Does it trigger an infinite loop? if yes we need to make sure it doesn't happen, our goal is to make the templates simpler.